artifacts/definitions/Windows/EventLogs/MessageHunter.yaml

```diff
@@ -85,13 +85,9 @@ sources:
                     EventTime > DateAfterTime 
                     AND format(format='%v',args=Message) =~ IocRegex
             })
+            GROUP BY EventID,Channel
```

Comment:
You need to take care when combining group by and order by - consider this case:

A|B
-|-
1|2
1|3
2|3
2|2

The group by returns the last row in each group:

A|B
-|-
1|3
2|2

Specifically for this query, if the query finds the same event in different sources, it will only give a source row for each event and so the order by will not be able to see multiple sources anyway (order by is applied **after** group by).  

So say you have event id 1 in C: and VSS1 and VSS2. The foreach query will return

source | event id
-|-
C: | 1
VSS1 | 1
VSS2 | 1

Then `group by event id` gives:

source| event id
-|-
VSS2 | 1

Since it drops each row it sees into a bin grouped by the same event id (in this case event id 1)

Now the order by is applied but it only has a single source anyway and will not go what you expect - we want to to return C: if there are two events.

I think we want to order by source on the files query so we parse the files in the reverse order - VSS1 and then C: 
This way when we hit the group by we ensure the last row is the C: one.

As a separate question - I think the intention is is to dedup the same event record from different VSS so I wonder if you mean to group by EventRecordID instead of EventId - as it stands this query will remove unrelated events with the same event types (for example multiple log on events will be merged - even for different times).

Reply:
Yep your right. In my testing there was only one event for each EID so I didnt pickup that typo. Previous versions it appears the invalid _group caused only one event in total so I fixed this case but introduced another.